src/Build/Graph/ProjectGraph.cs

```diff
@@ -106,5 +199,89 @@ private void LoadGraph(string entryProjectFile, ProjectCollection projectCollect
             }
         }
 
+        private static List<string> HydrateDefaultTargets(ProjectInstance project, List<string> targets)
+        {
+            // Avoid allocating until we have to
+            bool isMutated = false;
+
+            int i = 0;
+            while (i < targets.Count)
+            {
+                if (targets[i].Equals(DefaultTargetsMarker, StringComparison.OrdinalIgnoreCase))
+                {
+                    if (!isMutated)
+                    {
+                        // Make the copy now since we we're altering the list.
+                        targets = new List<string>(targets);
+                        isMutated = true;
+                    }
+
+                    targets.RemoveAt(i);
+
+                    var defaultTargets = project.DefaultTargets;
+                    for (int j = 0; j < defaultTargets.Count; j++)
+                    {
+                        targets.Insert(i++, defaultTargets[j]);
```

Comment:
List has InsertRange

Reply:
Good catch. This was originally an `IList` so I did this, but forgot about `InsertRange` when I decided to make it a concrete `List`